common/src/main/java/com/example/android/uamp/media/MusicService.kt

```diff
@@ -419,6 +449,23 @@ open class MusicService : MediaLibraryService() {
             customCommand: SessionCommand,
             args: Bundle
         ): ListenableFuture<SessionResult> {
+            /*
+            Toggle spatial audio
+             */
+            if (customCommand.customAction == ACTION_TOGGLE_SPATIALIZATION) {
+                val enable = customCommand.customExtras.getBoolean(EXTRAS_TOGGLE_SPATIALIZATION)
+
+                if (enable)
+                    uAmpAudioAttributesBuilder.setSpatializationBehavior(C.SPATIALIZATION_BEHAVIOR_AUTO)
+                else
+                    uAmpAudioAttributesBuilder.setSpatializationBehavior(C.SPATIALIZATION_BEHAVIOR_NEVER)
+
+                (exoPlayer as ExoPlayer).setAudioAttributes(
```

Comment:
It's a little odd to me that the Player interface doesn't expose a way of setting audio attributes. If that's an intentional decision by the media3 team, then I'd say it's better to change exoPlayer on line 141 to be an ExoPlayer instance, so that no casting is needed. Maybe Nevin has more context.

Reply:
Good point - I do believe it's intentional, since the `AudioAttributes` implementation being used here is specific to `ExoPlayer`, but I also don't see any reason why `exoPlayer` can't be an `ExoPlayer` instance, so +1 to making that change.